lib/cylc/config.py

```diff
@@ -1085,12 +1085,14 @@ def configure_queues(self):
         # Then reassign to other queues as requested.
         warnings = []
         requeued = []
-        for queue in queues:
-            if queue == 'default':
+        for key, queue in queues.copy().items():
+            # queues.copy() is essential here to allow items to be removed from
+            # the queues dict.
```

Comment:
OK understood, I hadn't realized you were looking ahead to Python 3.  We probably have many loops like this that will need changing in the forthcoming translation...